File: prototype/base/src/main/java/com/project/base/controller/AuthController.java
package com.project.base.controller;

import com.project.base.domain.dto.LoginRequest;
import com.project.base.domain.dto.SignUpRequest;
import com.project.base.domain.dto.TokenResponse;
import com.project.base.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final UserService userService;

    public AuthController(UserService userService) {
        this.userService = userService;
    }

    // 회원가입: 201 Created, 바디 없음
    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    public Mono<Void> signUp(@RequestBody SignUpRequest req) {
        return userService.signUp(req);
    }

    // 로그인: TokenResponse 반환
    @PostMapping("/login")
    public Mono<TokenResponse> login(@RequestBody LoginRequest req) {
        return userService.login(req.username(), req.password());
    }
}

File: prototype/lietzsche/src/main/kotlin/com/stxtory/lietzsche/controller/AuthController.kt
package com.stxtory.lietzsche.controller

import com.stxtory.lietzsche.dto.LoginRequest
import com.stxtory.lietzsche.dto.SignUpRequest
import com.stxtory.lietzsche.dto.TokenResponse
import com.stxtory.lietzsche.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Mono
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val userService: UserService,
) {

    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    fun signUp(@RequestBody req: SignUpRequest): Mono<Void> =
        userService.signUp(req.username, req.password).then()

    @PostMapping("/login")
    fun login(@RequestBody req: LoginRequest): Mono<TokenResponse> =
        userService.login(req.username, req.password)
}

@RestController
class MeController {
    @GetMapping("/api/me")
    fun me(@AuthenticationPrincipal principal: Any?): Mono<Map<String, Any?>> =
        Mono.just(mapOf("principal" to principal))
}

@ResponseStatus(HttpStatus.UNAUTHORIZED)
class InvalidCredentials : RuntimeException("invalid credentials")